tests/e2e/vmservice_vm.go

```diff
@@ -174,18 +236,386 @@ var _ bool = ginkgo.Describe("[vmsvc] vm service with csi vol tests", func() {
+		2	Create a PVC say pvc1
+		3	Create two VMservice VMs, say vm1 and vm2
+		4	Attach pvc1 to vm1
+		5	Try to attach pvc1 to vm2 which should fail
+		6	Delete VM service VMs
+		7	Delete pvc1
+		8	Remove spbm policy attached to test namespace
+
+		attach PVC used by one VM (powered off) to another VM
+		Steps:
+		1	Assign a spbm policy to test namespace with sufficient quota
+		2	Create a PVC say pvc1
+		3	Create two VMservice VMs, say vm1 and vm2
+		4	Attach pvc1 to vm1
+		5	power off vm1
+		6	Try to attach pvc1 to vm2 which should fail
+		7	Delete VM service VMs
+		8	Delete pvc1
+		9	Remove spbm policy attached to test namespace
+
+		attach PVC used by one VM to another VM (powered off)
+		Steps:
+		1	Assign a spbm policy to test namespace with sufficient quota
+		2	Create a PVC say pvc1
+		3	Create two VMservice VMs, say vm1 and vm2
+		4	Attach pvc1 to vm1
+		5	power off vm2
+		6	Try to attach pvc1 to vm2 which should fail
+		7	Power on vm2 and verify it comes up fine
+		8	Delete VM service VMs
+		9	Delete pvc1
+		10	Remove spbm policy attached to test namespace
+	*/
+	ginkgo.It("attach PVC used by one VM to another VM while in use", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		ginkgo.By("Create a storageclass")
+		storageclass, err := client.StorageV1().StorageClasses().Get(ctx, storageClassName, metav1.GetOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		ginkgo.By("Create a PVC")
+		pvc1, err := createPVC(client, namespace, nil, "", storageclass, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		pvc2, err := createPVC(client, namespace, nil, "", storageclass, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		pvcs := []*v1.PersistentVolumeClaim{pvc1, pvc2}
+		ginkgo.By("Waiting for all claims to be in bound state")
+		pvs, err := fpv.WaitForPVClaimBoundPhase(client, pvcs, pollTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			ginkgo.By("Delete PVCs")
+			err = fpv.DeletePersistentVolumeClaim(client, pvc1.Name, namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			err = fpv.DeletePersistentVolumeClaim(client, pvc2.Name, namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			ginkgo.By("Waiting for CNS volumes to be deleted")
+			err = e2eVSphere.waitForCNSVolumeToBeDeleted(pvs[0].Spec.CSI.VolumeHandle)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			err = e2eVSphere.waitForCNSVolumeToBeDeleted(pvs[1].Spec.CSI.VolumeHandle)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Creating VM bootstrap data")
+		secretName := createBootstrapSecretForVmsvcVms(ctx, client, namespace)
+		defer func() {
+			ginkgo.By("Deleting VM bootstrap data")
+			err := client.CoreV1().Secrets(namespace).Delete(ctx, secretName, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+		ginkgo.By("Creating VMs")
+		vm1 := createVmServiceVmWithPvcs(
```

Comment:
We can put this in a loop

Reply:
for 2 VMs it is fine, if I need more, will add a loop